### PR TITLE
Add unit test for Get_Reporting_Payment_Activity.

### LINKED ISSUE #8779: Create unit test for Get_Reporting_Payment_Activity
When reviewing https://github.com/Automattic/woocommerce-payments/pull/8743, I realized we don't have yet unit test for [Get_Reporting_Payment_Activity](https://github.com/Automattic/woocommerce-payments/blob/a4c1b31d4a4be6059bf91f31d8829bea1b180a3f/includes/core/server/request/class-get-reporting-payment-activity.php).

We can look at [Get_Charge_Test](https://github.com/Automattic/woocommerce-payments/blob/a4c1b31d4a4be6059bf91f31d8829bea1b180a3f/tests/unit/core/server/request/test-class-core-get-charge-request.php) as an inspiration.

### PR DESCRIPTION
**Pull request by shendy-a8c**
Fixes https://github.com/Automattic/woocommerce-payments/issues/8779

#### Changes proposed in this Pull Request

Add unit test for [Get_Reporting_Payment_Activity](https://github.com/Automattic/woocommerce-payments/blob/a4c1b31d4a4be6059bf91f31d8829bea1b180a3f/includes/core/server/request/class-get-reporting-payment-activity.php).

#### Testing instructions

* Make sure all unit tests pass.
* Make sure I cover important cases in the newly added unit test `Get_Reporting_Payment_Activity_Test` and I don't have copy-paste errors.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8785` or branch name `add/8779-unit-test-get-reporting-payment-activity` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/8779-unit-test-get-reporting-payment-activity) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 51b5121fab8a61969f84586fc951c7be1acb685c
- Build time: 2024-05-20 15:45:46 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: tests/unit/core/server/request/test-class-core-get-reporting-payment-activity-request.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+class Get_Reporting_Payment_Activity_Test extends WCPAY_UnitTestCase {
+
+	/**
+	 * Mock WC_Payments_API_Client.
+	 *
+	 * @var WC_Payments_API_Client|MockObject
+	 */
+	private $mock_api_client;
+	/**
+	 * Mock WC_Payments_API_Client.
```
Comment (Jinksi) [inline tests/unit/core/server/request/test-class-core-get-reporting-payment-activity-request.php]: ⛏️ code comment correction
```suggestion
	 * Mock WC_Payments_Http_Interface.
```

**Review by Jinksi — APPROVED** [review]
LGTM @shendy-a8c 🙌

💭 It may be worth adding a test for `set_date_end` _in addition to_ `set_date_start` date validation, but this is low-risk since they use the same validation logic.

**Comment by shendy-a8c** [comment]
Thank you for the review @Jinksi.

Good suggestion to cover date end separately. Addde in [783e538](https://github.com/Automattic/woocommerce-payments/pull/8785/commits/783e538b7b23aa175078a0c62d676f50cb27850b).

**Pull request merged**